azure/durable_functions/entity.py

```diff
@@ -0,0 +1,114 @@
+from .models import DurableEntityContext
+from .models.entities import OperationResult, EntityState
+from datetime import datetime
+from typing import Callable, Any, List, Dict
+
+
+class Entity:
+    """Durable Entity Class.
+
+    Responsible for execuitng the user-defined entity function.
+    """
+
+    def __init__(self, entity_func: Callable[[DurableEntityContext], None]):
+        """Create a new entity for the user-defined entity.
+
+        Responsible for executing the user-defined entity function
+
+        Parameters
+        ----------
+        entity_func: Callable[[DurableEntityContext], Generator[Any, Any, Any]]
+            The user defined entity function
+        """
+        self.fn: Callable[[DurableEntityContext], None] = entity_func
+
+    def handle(self, context: DurableEntityContext, batch: List[Dict[str, Any]]) -> str:
+        """Handle the execution of the user-defined entity function.
+
+        Loops over the batch, which serves to specify inputs to the entity,
+        and collects results and generates a final state, which are returned.
+
+        Parameters
+        ----------
+        context: DurableEntityContext
+            The entity context of the entity, which the user interacts with as their Durable API
+
+        Returns
+        -------
+        str
+            A JSON-formatted string representing the output state, results, and exceptions for the
+            entity execution.
+        """
+        response = EntityState(results=[], signals=[])
+        for packet in batch:
```

Comment:
`packet` as a name is very broad. I would prefer `operation`. I realize that is overloaded on `DurableEntityContext` to represent the operation name, but I still think it would be preferable as a reader.

Reply:
If I recall correctly, I think I chose the name `packet`, as you point out, to differentiate it from the `operation` in `DurableEntityContext`. I would like to avoid overloading these terms, because I think they can be confusing for a new reader trying to grasp how entities are implemented.

I've renamed the variable to `operation_data`, which I think is more accurate. What do you think?

Reply:
Seems like a happy compromise.